File: Vagharsh_Hakobyan/Homework/20151107/battle.cpp
#include <iostream>

int right_neighbor(int** arr, int i, int j, int size)
{
  for (int k=j; k<size; ++k)
  {
      if (arr[i][k]==0 || k==(size-1))
         {
            return k;
         }
  }
}

bool up_neighbor(int** arr, int i, int j)
{
  if (i==0 || arr[i-1][j]==0)
     {
         return true;
     }
  return false;
}

int main()
{
  int field = -1;
  do
  {
    std::cout << "Please iput size of field n = ";
    std::cin >> field;
  }
  while (field < 2);

  int **a;
  a = new int *[field];
  for (int i =0; i<field; i ++)
        a[i] = new int [field];
  for (int i=0;i<field;++i)
  {
      for (int j=0;j<field;++j)
      {
         std::cout << "Please input a[" << i+1 << "][" << j+1 << "]= ";
         std::cin >> a[i][j];
      }
  }
  int count=0;
  for (int i=0;i<field;++i)
  {
      for (int j=0;j<field;++j)
      {
          if (a[i][j]==1 && up_neighbor(a, i, j))
          {
             j=right_neighbor(a, i, j, field);
             count=count+1;
          }
      }
  }
  std::cout << "Count of boat is equal " << count << std::endl;
  for (int i = 0; i<field; i ++)
        delete a[i];
  delete a;
  return 0;
}

File: Vagharsh_Hakobyan/Homework/java/20160415/Battle.java
import java.util.Scanner;
class Battle {
	public static void main(String args[]){
		Scanner sc = new Scanner(System.in);
		int field = -1;
		do
		{
			System.out.print("Please iput size of field n = ");
			field = sc.nextInt();
		}
		while (field < 2); 
		
		int[][] a;
		a = new int[field][field];
		int i1 = 0;
		int j1 = 0;
		
		for (int i=0;i<field;++i)
			{
				for (int j=0;j<field;++j)
					{
						i1 = i+1;
						j1 = j+1;
						System.out.print("Please input a[" + i1 + "][" + j1 + "]= ");
						a[i][j] = sc.nextInt();
					}
			}
			
		int count=0;
		for (int i=0;i<field;++i)
			{
				for (int j=0;j<field;++j)
					{
						if (a[i][j]==1)
						{
							if (i==0 || a[i-1][j]==0)
							{
							for (int k=j; k< field; ++k)
								{
									if (a[i][k]==0 )
									{
										j=k;
									}
								}
							count=count+1;
							}
						}
					}
			}
	System.out.println("Count of boat is equal " + count);
	}
}
